packages/data/src/world/mm/great_bay_temple.yml

```diff
@@ -35,24 +35,24 @@
   dungeon: GB
   exits:
     "Great Bay Temple Water Wheel": "true"
-    "Great Bay Temple Map Room": "has_mask_zora || hookshot_anywhere || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))"
-    "Great Bay Temple Red Pipe 1": "has_mask_zora || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))"
+    "Great Bay Temple Map Room": "underwater_walking || hookshot_anywhere || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))"
+    "Great Bay Temple Red Pipe 1": "underwater_walking || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))"
     "Great Bay Temple Green Pipe 1": "can_use_ice_arrows || short_hook_anywhere"
-    "Great Bay Temple Compass Room": "(has_mask_zora || hookshot_anywhere || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))) && event(GB_WATER_WHEEL)"
-    "Great Bay Temple Pre-Boss": "(has_mask_zora || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))) && event(GB_WATER_WHEEL)"
+    "Great Bay Temple Compass Room": "(underwater_walking || hookshot_anywhere || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))) && event(GB_WATER_WHEEL)"
+    "Great Bay Temple Pre-Boss": "(underwater_walking || (trick(MM_GBT_CENTRAL_GEYSER) && ((can_use_fire_arrows && can_use_ice_arrows) || can_use_din || can_use_farore || can_use_nayru))) && event(GB_WATER_WHEEL)"
   events:
     MAGIC: "true"
   locations:
     "Great Bay Temple SF Central Room Barrel": "true"
-    "Great Bay Temple SF Central Room Underwater Pot": "has_mask_zora || (has_arrows && has(MASK_GREAT_FAIRY))"
+    "Great Bay Temple SF Central Room Underwater Pot": "underwater_walking || (has_arrows && has(MASK_GREAT_FAIRY))"
```

Comment:
is link able to roll into this or pick it up while being dragged by the currents? Or does he not get dragged when wearing the boots?

Reply:
Tried this for a while and wasn't able to get it. I'll change it to `has_mask_zora || (has_arrows && (has(MASK_GREAT_FAIRY) || underwater_walking_strict))` with strict coz it's possible but may take a while?

Reply:
Blast Mask would work. it could call for either that or strict i suppose.